src/solvers/td3.ml

```diff
@@ -1036,6 +1044,13 @@ module WP =
           )
 
         let finalize ~vh ~reachable =
+          (* replay superstable messages from unknowns that are still reachable *)
+          if incr_verify then (
+            HM.iter (fun (l:S.v) m ->
+                if HM.mem reachable l then Messages.add m
```

Comment:
Seems like `var_messages` should also be pruned by `IncrPrune` if this check is necessary. Otherwise these old messages still accumulate and are marshaled, but never printed again.